src/core/Global.ttslua

```diff
@@ -456,358 +268,281 @@ function drawChaostoken(params)
-function getDataTable(storage)
-    if CACHE.data[storage] == nil then
-        local obj = getObjectCache(tokenDataId)
+    if value.append ~= nil then
+      for _, token in ipairs(value.append) do
+        local obj = spawnChaosToken(token, pos)
         if obj ~= nil then
-            CACHE.data[storage] = obj.getTable(storage)
+          chaosbag.putObject(obj)
         end
+      end
     end
-    return CACHE.data[storage]
-end
 
-function getDataValue(storage, key)
-    local data = getDataTable(storage)
-    if data ~= nil then
-        local value = data[key]
-        if value ~= nil then
-            local res = {}
-            for m, v in pairs(value) do
-                res[m] = v
-                if res[m].parent ~= nil then
-                    local parentData = getDataValue(storage, res[m].parent)
-                    if parentData ~= nil and parentData[m] ~= nil and parentData[m].token ~= nil then
-                        res[m].token = parentData[m].token
-                    end
-                    res[m].parent = nil
-                end
-            end
-            return res
+    -- randomly choose tokens for specific Carcosa scenarios in standalone
+    if value.random then
+      local n = #value.random
+      if n > 0 then
+        for _, token in ipairs(value.random[math.random(1, n)]) do
+          local obj = spawnChaosToken(token, pos)
+          if obj ~= nil then
+            chaosbag.putObject(obj)
+          end
         end
+      end
     end
-end
 
-function getRandomCount(to)
-    updateRandomSeed()
-    return math.random(1, to)
+    if value.message then broadcastToAll(value.message) end
+    if value.warning then broadcastToAll(value.warning, { 1, 0.5, 0.5 }) end
+  end
 end
 
-function updateRandomSeed()
-    local chance = math.random(1,10)
-    if chance == 1 then
-        math.randomseed(os.time())
+function getDataValue(storage, key)
+  local data = getObjectFromGUID(tokenDataId).getTable(storage)
+  if data ~= nil then
+    local value = data[key]
+    if value ~= nil then
+      local res = {}
+      for m, v in pairs(value) do
+        res[m] = v
+        if res[m].parent ~= nil then
+          local parentData = getDataValue(storage, res[m].parent)
+          if parentData ~= nil and parentData[m] ~= nil and parentData[m].token ~= nil then
+            res[m].token = parentData[m].token
+          end
+          res[m].parent = nil
+        end
+      end
+      return res
     end
+  end
 end
 
+function spawnChaosToken(id, pos)
+  local url = getImageUrl(id)
+  if url ~= '' then
+    local obj = spawnObject({
+      type = 'Custom_Tile',
+      position = {pos.x, pos.y + 3, pos.z},
+      rotation = {0, 260, 0}
+    })
+    obj.setCustomObject({
+      type = 2,
+      image = url,
+      thickness = 0.1
+    })
+    obj.scale {0.81, 1, 0.81}
+    obj.setName(getTokenName({ url=url }))
+    return obj
+  end
+end
 
--- Content Importing
+-- chaos bag needs this!
+function getTokenName(params)
```

Comment:
Rename to getChaosTokenName, add doc for paramaters

Reply:
sadly this is also referenced by some setup bags (like for example from the custom campaign "Jumanji")